source/esm.civet

```diff
@@ -21,17 +21,42 @@ Previously depended on ts-node esm loader being downstream:
 ```bash
 node --loader ts-node/esm --loader @danielx/civet/esm source.civet
 ```
+
+If you want to configure this ESM loader, you can make your own
+`register.mjs` along these lines:
+
+@example
+```javascript
+import { register } from 'node:module';
+register('@danielx/civet/esm', import.meta.url, {data: {
+  parseOptions: {
+    // Add your parse options here
+  },
+}});
+```
+
 */
 
-{ readFileSync } from fs
+{ readFile } from fs/promises
+{ dirname } from path
 { pathToFileURL, fileURLToPath } from url
 
-Civet from ./main.civet
-{ compile, SourceMap } := Civet
+{ compile, SourceMap } from ./main.civet
+{ findConfig, loadConfig } from ./config.civet
+type { ParseOptions } from @danielx/civet
+
+export type RegisterOptions
+  config?: string | null
```

Comment:
I'd prefer keeping this api as `string | false`. I think `config: false` reads slightly better than `config: null` in terms of intent.

Reply:
I'm fine with changing to/allowing `false`, though we should probably continue to support `null` because that's the current API for the unplugin (the only place where `config` can currently be specified):

https://github.com/DanielXMoore/Civet/blob/0039cdd9c0351815adf810eb2f3c39276472b450/source/unplugin/unplugin.civet#L46-L47

Maybe `| false | null` for all of them?

Reply:
:+1: Allowing both is fine.

Reply:
Reviewing the code closer, the unplugin actually supported `config: undefined` to disable config files. I think this is a bit too much: it's easy to do something like `{config: someOtherObject?.config}` and accidentally set `undefined` for an absent property. So I refactored a bit to treat `undefined` just like an unspecified property, i.e., to search for a config file, matching the documentation.

Anyway, now we support `config: false` and `config: null` (and probably any falsey value other than `undefined`), and it should be the same in the unplugin and the ESM loader (and the CLI too, though there it's just an internal notion of `config: false`, now restored).